static_frame/core/index_hierarchy.py

```diff
@@ -1399,6 +1399,41 @@ def values_at_depth(self: IH,
             return self._blocks._extract_array_column(depth_level)
         return self._blocks._extract_array(column_key=list(depth_level))
 
+    def index_at_depth(self: IH,
+            depth_level: DepthLevelSpecifier = 0,
+            ) -> tp.Union[Index, tp.List[Index]]:
+        '''
+        Return an index, or a list of indices for the ``depth_level`` specified.
+
+        Args:
+            depth_level: a single depth level, or iterable depth of depth levels.
+        '''
+        if self._recache:
+            self._update_array_cache()
+
+        if isinstance(depth_level, INT_TYPES):
+            return self._indices[depth_level]
+
+        return list(map(self._indices.__getitem__, depth_level))
```

Comment:
Any specific reason to return a list over a tuple?
```suggestion
        return tuple(map(self._indices.__getitem__, depth_level))
```